Solutions/Marain.UserNotifications.Management.Host/Marain/UserNotifications/Management/Host/Composer/GenerateTemplateComposer.cs

```
@@ -0,0 +1,136 @@
+
+                            emailTemplate = new EmailTemplate()
+                            {
+                                NotificationType = notificationType,
+                                Body = emailBody,
+                                Subject = emailSubject,
+                                Important = false,
+                            };
+                        }
+                        catch (Exception)
+                        {
+                            this.logger.LogError("The template for the communication type Email doesn't exist");
+                        }
+
+                        break;
+                    case CommunicationType.Sms:
+                        try
+                        {
+                            (SmsTemplate, string?) smsRawTemplate = await templateStore.GetAsync<SmsTemplate>(notificationType, CommunicationType.Sms).ConfigureAwait(false);
+                            string? smsBody = await this.GenerateTemplateForFieldAsync(smsRawTemplate.Item1.Body!, existingProperties).ConfigureAwait(false);
+
+                            smsTemplate = new SmsTemplate() { NotificationType = notificationType, Body = smsBody };
+                        }
+                        catch (Exception)
+                        {
+                            this.logger.LogError("The template for the communication type Sms doesn't exist");
+                        }
+
+                        break;
+                    case CommunicationType.WebPush:
+                        try
+                        {
+                            (WebPushTemplate, string?) webPushRawTemplate = await templateStore.GetAsync<WebPushTemplate>(notificationType, CommunicationType.WebPush).ConfigureAwait(false);
+                            string? webPushTitle = await this.GenerateTemplateForFieldAsync(webPushRawTemplate.Item1.Title, existingProperties).ConfigureAwait(false);
```

With this and the other cases, shouldn't it throw an exception if certain fields aren't produced? It doesn't seem wise to swallow an exception and send a Web Push notification with no body, or an email with no subject.

---

So I didn't want the class to throw an exception as this composer will generate the template for all the 3 communication types. Thus, I have logged the exceptions if the body or subject is null and the WebPush / EmailTemplate will be null and the delivery channel would not trigger a notification for the unconfigured communication type and would carry to send for the ones which are configured.